File: AdminPanel1/AdminPanel1.Web/Modules/ServerTypes/SouqCom/CartColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { CartRow } from "./CartRow";

export interface CartColumns {
    Id: Column<CartRow>;
    UserId: Column<CartRow>;
    ProductName: Column<CartRow>;
    Qty: Column<CartRow>;
}

export class CartColumns extends ColumnsBase<CartRow> {
    static readonly columnsKey = 'SouqCom.Cart';
    static readonly Fields = fieldsProxy<CartColumns>();
}

File: AdminPanel1/AdminPanel1.Web/Modules/SouqCom/Cart/CartColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace AdminPanel1.SouqCom.Columns;

[ColumnsScript("SouqCom.Cart")]
[BasedOnRow(typeof(CartRow), CheckNames = true)]
public class CartColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string UserId { get; set; }
    public string ProductName { get; set; }
    public int Qty { get; set; }
}